frontend/src/shared/components/Spacing.tsx

```diff
@@ -1,9 +1,64 @@
-import { styled } from 'styled-components';
+import styled, { css } from 'styled-components';
+import type { BreakPoints } from '@/shared/types/theme';
 
-const Spacing = styled.div<{ direction: 'horizontal' | 'vertical'; size: number }>`
+interface Spacing {
+  direction: 'horizontal' | 'vertical';
+  size: number;
+}
+
+interface ResponsiveSpacing extends Partial<Record<`$${BreakPoints}`, Partial<Spacing>>> {}
+
+interface SpacingProps extends Spacing, ResponsiveSpacing {}
+
+const Spacing = styled.div<SpacingProps>`
   flex: none;
   min-width: ${({ direction, size }) => (direction === 'horizontal' ? `${size}px` : '0')};
   min-height: ${({ direction, size }) => (direction === 'vertical' ? `${size}px` : '0')};
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.xxl}) {
+    ${({ $xxl, direction, size }) => spacingCss($xxl, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.xl}) {
+    ${({ $xl, direction, size }) => spacingCss($xl, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.lg}) {
+    ${({ $lg, direction, size }) => spacingCss($lg, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.md}) {
+    ${({ $md, direction, size }) => spacingCss($md, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.sm}) {
+    ${({ $sm, direction, size }) => spacingCss($sm, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.xs}) {
+    ${({ $xs, direction, size }) => spacingCss($xs, direction, size)}
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakPoints.xxs}) {
+    ${({ $xxs, direction, size }) => spacingCss($xxs, direction, size)}
+  }
 `;
 
+const spacingCss = (
+  spacing?: Partial<Spacing>,
+  originalDir?: Spacing['direction'],
+  originalSize?: Spacing['size']
+) => {
+  if (!spacing) return;
+  const { direction: newDirection, size: newSize } = spacing;
+
+  const realDirection = newDirection ?? originalDir;
+  const realSize = newSize ?? originalSize;
+
+  return css`
+    min-width: ${realDirection === 'horizontal' ? `${realSize}px` : '0'};
+    min-height: ${realDirection === 'vertical' ? `${realSize}px` : '0'};
+  `;
+};
```

Comment:
💬 해당 함수를 이해하는게 힘들었어용..🥲
real은 뭐지.. new는 뭐지.. 하다가 아래와 같이 제안드립니다.

```suggestion
const spacingCss = (
  responsiveSpacing?: Partial<Spacing>,
  originalDir?: Spacing['direction'],
  originalSize?: Spacing['size']
) => {
  if (!responsiveSpacing) return;

  const direction = responsiveSpacing.direction ?? originalDir;
  const size = responsiveSpacing.size ?? originalSize;

  return css`
    min-width: ${direction === 'horizontal' ? `${size}px` : '0'};
    min-height: ${direction === 'vertical' ? `${size}px` : '0'};
  `;
};
```

Reply:
코드 설명을 달아놨어야 하는데, 깜빡했네요!!
저도 코난 의견에 동의합니다! 
Flex 컴포넌트에 적용된 방식대로 인자명을 비슷하게 가져가려고 하다보니 읽기 어려워졌네요!

적극 반영하였습니다~